src/Domain.LinnApps/StockLocators/StockLocatorService.cs

```diff
@@ -0,0 +1,122 @@
+﻿namespace Linn.Stores.Domain.LinnApps.StockLocators
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Stores.Domain.LinnApps.Exceptions;
+
+    public class StockLocatorService : IStockLocatorService
+    {
+        private readonly IRepository<StockLocator, int> repository;
+
+        private readonly IRepository<StoresPallet, int> palletRepository;
+
+        private readonly IQueryRepository<StoragePlace> storagePlaceRepository;
+
+        public StockLocatorService(
+            IRepository<StockLocator, int> repository,
+            IRepository<StoresPallet, int> palletRepository,
+            IQueryRepository<StoragePlace> storagePlaceRepository)
+        {
+            this.repository = repository;
+            this.palletRepository = palletRepository;
+            this.storagePlaceRepository = storagePlaceRepository;
+        }
+
+        public void UpdateStockLocator(StockLocator @from, StockLocator to)
```

Comment:
Is this @ a typo?

Reply:
No, I think from is a reserved word and so the @ is needed to escape it? Not sure. Could just rename the parameter.